hw9/lesson9.py

```diff
@@ -0,0 +1,71 @@
+num_fib = int(input('Enter the serial number of the number from the Fibonacci sequence: '))
+print('\n')
+# generator
+
+
+def gen_fib(num_fib):
+    fib_1, fib_2 = 0, 1
+    for i in range(num_fib+1):
+        yield fib_1
+        fib_1, fib_2 = fib_2, fib_1 + fib_2
+
+c = -1
+for i in gen_fib(num_fib):
+    c += 1
+    if num_fib == c:
+        print('Fibonacci sequence using Generator: ')
+        print(f'The {num_fib}th number in the Fibonacci sequence is {i}\n')
+
+# iterator
+
+
+class IterFib:
+    def __init__(self):
+        self.current = 0
+        self.next = 1
+
+    def __iter__(self):
+        return self
+
+    def __next__(self):
+        temp = self.current
```

Comment:
There is no raising of StopIteraton; in some cases, it could be dangerous. For example, using this iterator in `for in` construction will generate values endlessly. And in most cases, iterators are using in `for in` but not by calling `next` function

Reply:
Alexey, I have added raise StopIteration, could you validate, please